Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,218 @@
+  (scope) AllUsers or CurrentUser.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or Long Term Service (LTS) release. Preview should not be allowed if value of
+  parameter **-Version** is Generally Available (GA).
+- Parameter: Passthru - to provide informative output post installation
+
+## Syntax and Parameter Sets
+
+### ParameterSet Version (Default)
+
+```
+Install-PowerShell7 [[-Version] <string>] [-Scope {AllUsers | CurrentUsers}] [-Passthru] [-WhatIf] [-Confirm]  [<CommonParameters>]
+```
+
+### ParameterSet Channel
+
+```
+Install-PowerShell7 -Channel {Preview | LTS | Stable} [-Scope {AllUsers | CurrentUser}] [-Passthru] [-WhatIf] [-Confirm]
+[<CommonParameters>]
+```
+
+The **Version** parameter has the following validation attribute: [ValidatePattern("^7\.\d+")]
+
+The **Channel** parameter has the following validation attribute: [ValidateSet("Preview", "LTS", "Stable")]
+
+The **Scope** parameter has the following validation attribute: [ValidateSet("AllUsers", "CurrentUser")]
+
+### Data model
+
+The `Install-PowerShell7` cmdlet should support the **Passthru** parameter to provide useful
+information post installation. The data model for the output is below;
+
+```output
+PSTypename   = "PSInstallationInfo"
+[Version]Version      = $version
+[string]Path         = (Get-Command pwsh).source
+[string]Channel      = if ($Channel) { $channel} else {"Stable" }
+[string]Computername = $env:COMPUTERNAME
+```
+
+## Demo.txt
+
+- Install the latest stable version by default.
+
+  ```powershell
+  install-powershell7 -Passthru
+  ```
+
+  ```
+  Version Path                                   Channel Computername
+  ------- ----                                   ------- ------------
+  7.2.3   C:\Program Files\PowerShell\7\pwsh.exe Stable
+  ```
+
+- To install the latest stable release of PowerShell over PowerShell Remoting:
+
+  ```powershell
+  Invoke-Command -Computername RemoteComputer1 -ScriptBlock {Install-PowerShell7}
+  ```
+
+- Install the LTS release using the current user scope.
+
+  ```powershell
+  Install-PowerShell7 -Passthru -Channel LTS -Scope CurrentUser
+  ```
+
+  ```
+  Version Path                                   Channel Computername
+  ------- ----                                   ------- ------------
+  7.2.0   C:\Users\jeff\PowerShell\7\pwsh.exe    LTS     RemoteComp
+  ```
+
+- To install the latest stable release of PowerShell:
+
+  ```powershell
+  Install-PowerShell7
+  ```
+
+- To install a specific version of a PowerShell release:
+
+  ```powershell
+  Install-PowerShell7 -Version 7.2
+  ```
+
+- To install a specific version of a PowerShell release to a specific scope:
+
+  ```powershell
+  Install-PowerShell7 -Version 7.2 -Scope Allusers
+  ```
+
+- To install the latest preview from a specific version:
+
+  ```powershell
+  Install-PowerShell7 -Version 7.3 -Channel 'Preview'
```

Comment:
couple of questions:
- will the msi/msix/zip file removed (and will we check a signature?)
- there may be situations where the file can't be downloaded (or installed after downloading) what will those errors look like?
- are there other error conditions which should be enumerated?